plugins/userdirs/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+use std::fs::File;
+
+use std::io::{self, BufRead};
+use std::path::Path;
+use std::process::Command;
+
+use fuzzy_matcher::FuzzyMatcher;
+
+use abi_stable::std_types::{ROption, RString, RVec};
+use anyrun_plugin::*;
+
+pub struct State {
+    dirs: Vec<String>,
+}
+
+#[init]
+fn init(_config_dir: RString) -> State {
+    let mut dirs: Vec<String> = vec![];
+
+    match home::home_dir() {
+        Some(path) => {
+            if let Ok(lines) = read_lines(format!("{}/.config/user-dirs.dirs", path.display())) {
```

Comment:
Not sure if this is present on every system. It is present on mine (auto-generated by nix).